x-pack/plugins/maps/public/actions/map_actions.js

```diff
@@ -111,14 +111,36 @@ function getLayerById(layerId, state) {
   });
 }
 
-async function syncDataForAllLayers(dispatch, getState, dataFilters) {
-  const state = getState();
-  const layerList = getLayerList(state);
-  const syncs = layerList.map(layer => {
-    const loadingFunctions = getLayerLoadingCallbacks(dispatch, getState, layer.getId());
-    return layer.syncData({ ...loadingFunctions, dataFilters });
-  });
-  await Promise.all(syncs);
+function syncDataForAllLayers() {
+  return async (dispatch, getState) => {
+    const syncPromises = getLayerList(getState()).map(async layer => {
+      return dispatch(syncDataForLayer(layer));
+    });
+    await Promise.all(syncPromises);
+  };
+}
+
+function syncDataForLayer(layer) {
+  return async (dispatch, getState) => {
+    const dataFilters = getDataFilters(getState());
+    if (!layer.isVisible() || !layer.showAtZoomLevel(dataFilters.zoom)) {
```

Comment:
thanks for pulling this out here. it really helps when the business-logic is gathered in the actions rather than sprinkled across multiple classes.